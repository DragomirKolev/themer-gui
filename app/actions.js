// Actions

export const COLOR_CHANGE = 'COLOR_CHANGE';
export const FOCUS_MODE_TOGGLE = 'FOCUS_MODE_TOGGLE ';
export const EXPORT_DIALOG_OPEN = 'EXPORT_DIALOG_OPEN';
export const HELP_DIALOG_OPEN = 'HELP_DIALOG_OPEN';
export const CLOSE_DIALOGS = 'CLOSE_DIALOGS';
export const SET_EXPORT_OPTION = 'SET_EXPORT_OPTION';
export const EXPORT_THEMES = 'EXPORT_THEMES';

// Action generators

export const colorChange = (colorSetKey, colorKey, value) => ({
  type: COLOR_CHANGE,
  colorSetKey,
  colorKey,
  value,
});

export const focusModeToggle = (isLight) => ({
  type: FOCUS_MODE_TOGGLE,
  isLight,
});

export const exportDialogOpen = () => ({
  type: EXPORT_DIALOG_OPEN,
});

export const helpDialogOpen = () => ({
  type: HELP_DIALOG_OPEN,
});

export const closeDialogs = () => ({
  type: CLOSE_DIALOGS,
});

export const setExportOption = (option, value) => ({
  type: SET_EXPORT_OPTION,
  option,
  value,
});

export const exportThemes = () => ({
  type: EXPORT_THEMES,
});
