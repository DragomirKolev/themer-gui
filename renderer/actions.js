// Actions

export const COLOR_CHANGE = 'COLOR_CHANGE';
export const FOCUS_MODE_TOGGLE = 'FOCUS_MODE_TOGGLE ';
export const EXPORT_DIALOG_OPEN = 'EXPORT_DIALOG_OPEN';
export const TIPS_DIALOG_OPEN = 'TIPS_DIALOG_OPEN';
export const CLOSE_DIALOGS = 'CLOSE_DIALOGS';
export const SET_EXPORT_OPTION = 'SET_EXPORT_OPTION';
export const EXPORT_PROGRESS_DIALOG_OPEN = 'EXPORT_PROGRESS_DIALOG_OPEN';
export const EXPORT_CANCELLED = 'EXPORT_CANCELLED';
export const EXPORT_PROGRESS = 'EXPORT_PROGRESS';
export const EXPORT_ERROR = 'EXPORT_ERROR';
export const EXPORT_COMPLETE = 'EXPORT_COMPLETE';
export const EXPORT_RESET = 'EXPORT_RESET';
export const PREFILL_DIALOG_OPEN = 'PREFILL_DIALOG_OPEN';
export const PREFILL_COLOR_SET_SELECTION_CHANGE = 'PREFILL_COLOR_SET_SELECTION_CHANGE';
export const PREFILL_WITH_COLOR_SET = 'PREFILL_WITH_COLOR_SET';
export const SAVE_COMPLETE = 'SAVE_COMPLETE';
export const OPEN_COMPLETE = 'OPEN_COMPLETE';
export const RESET_STATE = 'RESET_STATE';
export const UPDATE_READY = 'UPDATE_READY';
export const NEXT_TIP = 'NEXT_TIP';
export const PREVIOUS_TIP = 'PREVIOUS_TIP';

// Action generators

export const colorChange = (colorSetKey, colorKey, value) => ({
  type: COLOR_CHANGE,
  colorSetKey,
  colorKey,
  value,
});

export const focusModeToggle = (isLight) => ({
  type: FOCUS_MODE_TOGGLE,
  isLight,
});

export const exportDialogOpen = () => ({
  type: EXPORT_DIALOG_OPEN,
});

export const tipsDialogOpen = () => ({
  type: TIPS_DIALOG_OPEN,
});

export const closeDialogs = () => ({
  type: CLOSE_DIALOGS,
});

export const setExportOption = (option, value) => ({
  type: SET_EXPORT_OPTION,
  option,
  value,
});

export const exportProgressDialogOpen = () => ({
  type: EXPORT_PROGRESS_DIALOG_OPEN,
});

export const exportCancelled = () => ({
  type: EXPORT_CANCELLED,
});

export const exportProgress = (status) => ({
  type: EXPORT_PROGRESS,
  status,
});

export const exportError = (message) => ({
  type: EXPORT_ERROR,
  message,
});

export const exportComplete = (exportedPath) => ({
  type: EXPORT_COMPLETE,
  exportedPath,
});

export const exportReset = () => ({
  type: EXPORT_RESET,
});

export const prefillDialogOpen = () => ({
  type: PREFILL_DIALOG_OPEN,
});

export const prefillColorSetSelectionChange = (selection) => ({
  type: PREFILL_COLOR_SET_SELECTION_CHANGE,
  selection,
});

export const prefillWithColorSet = (colorSet) => ({
  type: PREFILL_WITH_COLOR_SET,
  colorSet,
});

export const saveComplete = (filePath) => ({
  type: SAVE_COMPLETE,
  filePath,
});

export const openComplete = (fileData) => ({
  type: OPEN_COMPLETE,
  fileData,
});

export const resetState = () => ({
  type: RESET_STATE,
});

export const updateReady = () => ({
  type: UPDATE_READY,
});

export const nextTip = () => ({
  type: NEXT_TIP,
});

export const previousTip = () => ({
  type: PREVIOUS_TIP,
});
