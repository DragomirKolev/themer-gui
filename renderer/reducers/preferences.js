import { SET_SHOW_TIPS_ON_STARTUP, SET_SHOW_OVERWRITE_SHADES_WARNING } from '../actions';

export const defaultPreferences = {
  showTipsOnStartup: true,
  showOverwriteShadesWarning: true,
};

export const preferencesReducer = (state = defaultPreferences, action) => {
  switch (action.type) {
    case SET_SHOW_TIPS_ON_STARTUP:
      return {
        ...state,
        showTipsOnStartup: action.show,
      };
    case SET_SHOW_OVERWRITE_SHADES_WARNING:
      return {
        ...state,
        showOverwriteShadesWarning: action.show,
      };
    default:
      return state;
  }
};
