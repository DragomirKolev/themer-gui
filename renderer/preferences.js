import { defaultPreferences } from './reducers/preferences';

const PREF_KEY = 'preferences';

export const readPreferences = () => {
  try {
    return {
      ...defaultPreferences,
      ...JSON.parse(window.localStorage.getItem(PREF_KEY)),
    };
  }
  catch (e) {
    return defaultPreferences;
  }
};

const writePrefs = prefs => window.localStorage.setItem(PREF_KEY, JSON.stringify(prefs));

export const getPreference = key => readPreferences()[key];

export const connectPreferences = store => {
  store.subscribe(() => {
    writePrefs(store.getState().preferences);
  });
};
