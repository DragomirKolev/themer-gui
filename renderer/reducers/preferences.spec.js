import { preferencesReducer } from './preferences';
import { setShowTipsOnStartup, setShowOverwriteShadesWarning } from '../actions';

describe('preferencesReducer', () => {
  it('should initialize with showing tips on startup', () => {
    expect(preferencesReducer(undefined, {}).showTipsOnStartup).toBe(true);
  });
  it('should toggle showing tips on startup appropriately', () => {
    expect(preferencesReducer(undefined, setShowTipsOnStartup(false))).toMatchSnapshot();
    expect(preferencesReducer(undefined, setShowTipsOnStartup(true))).toMatchSnapshot();
  });
  it('should initialize with showing overwrite shades warning', () => {
    expect(preferencesReducer(undefined, {}).showOverwriteShadesWarning).toBe(true);
  });
  it('should toggle showing overwrite shades warning appropriately', () => {
    expect(preferencesReducer(undefined, setShowOverwriteShadesWarning(false))).toMatchSnapshot();
    expect(preferencesReducer(undefined, setShowOverwriteShadesWarning(true))).toMatchSnapshot();
  });
});
