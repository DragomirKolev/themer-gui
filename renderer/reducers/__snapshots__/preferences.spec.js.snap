// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preferencesReducer should toggle showing overwrite shades warning appropriately 1`] = `
Object {
  "showOverwriteShadesWarning": false,
  "showTipsOnStartup": true,
}
`;

exports[`preferencesReducer should toggle showing overwrite shades warning appropriately 2`] = `
Object {
  "showOverwriteShadesWarning": true,
  "showTipsOnStartup": true,
}
`;

exports[`preferencesReducer should toggle showing tips on startup appropriately 1`] = `
Object {
  "showOverwriteShadesWarning": true,
  "showTipsOnStartup": false,
}
`;

exports[`preferencesReducer should toggle showing tips on startup appropriately 2`] = `
Object {
  "showOverwriteShadesWarning": true,
  "showTipsOnStartup": true,
}
`;
